src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb.model.ts

```diff
@@ -26,3 +26,38 @@ export interface AuthMode {
   user: 'User';
   activeDirectory: 'active-directory';
 }
+
+export interface JoinAuth {
+  resource_type: string;
+  auth_id: string;
+  intent: 'present' | 'removed';
+  auth: Auth;
+  linked_to_cluster?: string;
+}
+
+export interface UsersGroups {
+  resource_type: string;
+  users_groups_id: string;
+  intent: 'present' | 'removed';
```

Comment:
I see this is being used multiple places, we can create a type or enum .

Reply:
Added type `Intent`